### PR TITLE
feat(pkgrepo): impl new logic after migration to `packages.broadcom.com`

### PR DESCRIPTION
**Pull request by sticky-note**
### PR progress checklist (to be filled in by reviewers)

- [x] Changes to documentation are appropriate (or tick if not required)
- [ ] Changes to tests are appropriate (or tick if not required)
- [ ] Reviews completed

---

#### Primary type

- [ ] `[build]`    Changes related to the build system
- [ ] `[chore]`    Changes to the build process or auxiliary tools and libraries such as documentation generation
- [ ] `[ci]`       Changes to the continuous integration configuration
- [x] `[feat]`     A new feature
- [ ] `[fix]`      A bug fix
- [ ] `[perf]`     A code change that improves performance
- [ ] `[refactor]` A code change that neither fixes a bug nor adds a feature
- [ ] `[revert]`   A change used to revert a previous commit
- [ ] `[style]`    Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)

#### Secondary type

- [ ] `[docs]`     Documentation changes
- [ ] `[test]`     Adding missing or correcting existing tests

### Does this PR introduce a `BREAKING CHANGE`?

No. The Formula was broken since the migration after all.

### Related issues and/or pull requests

Resolves: https://github.com/saltstack-formulas/salt-formula/issues/565
Obsoletes: https://github.com/saltstack-formulas/salt-formula/pull/561
Maybe can closes: https://github.com/saltstack-formulas/salt-formula/issues/545

### Describe the changes you're proposing

Update logic following the migration to `packages.broadcom.com`. For further info, see:
-> https://saltproject.io/blog/upcoming-salt-project-docs-and-repo-migration/
-> https://docs.saltproject.io/salt/install-guide/en/latest/

### Documentation checklist

- [ ] Updated the `README` (e.g. `Available states`).
- [x] Updated `pillar.example`.

### Testing checklist

- [ ] Included in Kitchen (i.e. under `state_top`).
- [ ] Covered by new/existing tests (e.g. InSpec, Serverspec, etc.).
- [ ] Updated the relevant test pillar.

### CODE REVIEW

File: pillar.example [hunk 1]
```diff
@@ -37,15 +37,15 @@ salt:
     salt_ssh: 'salt-ssh'
     pyinotify: 'python-pyinotify'  # the package to be installed for pyinotify
 
-  # Set which salt repository to use, default to https://repo.saltproject.io
-  # For older releases use https://archive.repo.saltproject.io
-  repo: 'https://archive.repo.saltproject.io'
+  # Set which salt repository to use, default to https://packages.broadcom.com/artifactory
+  repo: 'https://packages.broadcom.com/artifactory'
+  repo_key_url: 'https://packages.broadcom.com/artifactory/api/security/keypair/SaltProjectKey/public'
```
Comment (jeff350) [inline pillar.example:42]: This should fix the failure in the pipeline for the pre-commit job https://gitlab.com/saltstack-formulas/salt-formula/-/jobs/8356162035

```suggestion
  # yamllint disable rule:line-length
  repo: 'https://packages.broadcom.com/artifactory'
  repo_key_url: 'https://packages.broadcom.com/artifactory/api/security/keypair/SaltProjectKey/public'
  # yamllint enable rule:line-length
```

**Comment by sticky-note** [comment]
> @sticky-note Thank you for taking the time to put this together. I was able to give it a quick look at it now and noticed a few small things around the CI pipeline.
> 
> I added a suggestion above to fix the pre-commit job
> 
> The commit lint jobs are failing because the commit message is 6 characters longer than expected. Could you please amend the commit message to the expected length since this is parsed for the changelog.
> 
> ```
> Error: You have commit messages with errors
> 
> ⧗   input: feat(pkgrepo): update logic following the migration to `packages.broadcom.com`
> ✖   header must not be longer than 72 characters, current length is 78 [header-max-length]
> 
> ✖   found 1 problems, 0 warnings
> ⓘ   Get help: https://github.com/conventional-changelog/commitlint/#what-is-commitlint
> ```
> 
> https://github.com/saltstack-formulas/salt-formula/actions/runs/11830833414/job/32965007118?pr=566
> 
> I plan to take a closer look at this soon.

Done. Sorry for that. I'am actually Testing for Debian 12, Ubuntu 24.04, AlmaLinux 9, rocky 9
I will confirm if it works for these cases

Do I need to do something for SLES, OpenSUSE or the binaries are managed by them ?

File: salt/osfamilymap.yaml [hunk 2]
```diff
@@ -13,7 +13,8 @@
 {%- set osmajorrelease = salt['grains.get']('osmajorrelease', osrelease)|string %}
 {%- set oscodename = salt['grains.get']('oscodename') %}
```
Comment (sir-ragna) [inline salt/osfamilymap.yaml:14]: If the variables `osmajorrelease` and `oscodename` are no longer used they should be removed.

File: pillar.example [hunk 3]
```diff
@@ -37,15 +37,17 @@ salt:
     salt_ssh: 'salt-ssh'
     pyinotify: 'python-pyinotify'  # the package to be installed for pyinotify
 
-  # Set which salt repository to use, default to https://repo.saltproject.io
-  # For older releases use https://archive.repo.saltproject.io
-  repo: 'https://archive.repo.saltproject.io'
+  # Set which salt repository to use, default to https://packages.broadcom.com/artifactory
+  # yamllint disable rule:line-length
+  repo: 'https://packages.broadcom.com/artifactory'
+  repo_key_url: 'https://packages.broadcom.com/artifactory/api/security/keypair/SaltProjectKey/public'
```
Comment (jeff350) [inline pillar.example:43]: I actually put that in the wrong place on the first review. the disable rule needed to be before the comment on line 40. 

```suggestion
  # yamllint disable rule:line-length
  # Set which salt repository to use, default to https://packages.broadcom.com/artifactory
  repo: 'https://packages.broadcom.com/artifactory'
  repo_key_url: 'https://packages.broadcom.com/artifactory/api/security/keypair/SaltProjectKey/public'
```

Comment (sticky-note) [reply pillar.example]: Done.  I cut the line

File: salt/osfamilymap.yaml (hunk 2 above)
Comment (sticky-note) [reply salt/osfamilymap.yaml]: Done removed all unused variables

**Comment by jeff350** [comment]
Looking at the CI jobs it appears that `py_ver_repr` was removed somewhere where it shouldn't have been and it is now causing rendering issues in the SLS files.

https://gitlab.com/saltstack-formulas/salt-formula/-/jobs/8376038655

**Comment by sticky-note** [comment]
> Looking at the CI jobs it appears that `py_ver_repr` was removed somewhere where it shouldn't have been and it is now causing rendering issues in the SLS files.
> 
> https://gitlab.com/saltstack-formulas/salt-formula/-/jobs/8376038655

Corrected the mistyping, Sorry for that.

I confirm It is working on Ubuntu 22.04, Debian 12, Almalinux 9 with salt `3007.1`
Maybe, there are some work to do for the repo management on `RedHat` by this formula, maybe it can be done later.

Don't know how to test with SUSE, OpenSUSE

**Pull request merged**